tests/App/Admin/AuthorAdmin.php

```diff
@@ -40,15 +41,17 @@ protected function configureDatagridFilters(DatagridMapper $filter): void
     protected function configureFormFields(FormMapper $form): void
     {
         $form
-            ->add('id', null, [
+            ->add('id', TextType::class, [
                 'attr' => [
                     'class' => 'author_id',
                 ],
+                'empty_data' => '',
```

Comment:
Can you explain these changes ? Why is it needed for Sonata 4 ?

Reply:
I'll probably cherry-pick this commit in `3.x` (once I'll make this work). These changes are made to force to be a `string` when submitting the field as empty, otherwise it would submit a `null` value and the setters are defined like:

https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/6290d3f08f6df4c7858187bdb936d6e295c42862/tests/App/Document/Author.php#L68-L71

I haven't checked what has been changed in Sonata 4, but I guess it has to do with creating an associated document in a form.